core/request_handler.go

```diff
@@ -143,16 +148,22 @@ func (c *Connection) processRequest(ch *Channel, req *vppRequest) error {
 				"data_len": len(pingData),
 			}).Debugf(" -> SEND MSG: %T", c.msgControlPing)
 		}
-
-		t = time.Now()
-		if err := c.vppClient.SendMsg(context, pingData); err != nil {
+		// send the control ping request to VPP
+		if err = func() (err error) {
+			r := &api.Record{Message: c.msgControlPing, Timestamp: time.Now(), ChannelID: ch.id}
```

Comment:
This unnecessarily allocates record even in case the trace is not enabled. Which seems wasteful.

Reply:
Thanks, @ondrej-fabry, valid point. However, certain preallocations will be necessary... I cut it down to the bare minimum, the `registerNew` trace just "reserves" (creates) a time slot (and a place in the wait-group if enabled). This must be done before the `SendMsg()` is called, otherwise, the message order and expected record length will get unpredictable.
The record is now allocated only if the trace is enabled.